cleanlab/segmentation/rank.py

```diff
@@ -0,0 +1,240 @@
+      ``P(label=k|x)``. Each pixel contains an array of K classes, where for 
+      an example `x` the array at each pixel contains the model-predicted probabilities 
+      that `x` belongs to each of the K classes.
+      
+    method : {"softmin", "num_pixel_issues"}, default="softmin"
+      Label quality scoring method.
+
+    batch_size : int, optional
+      For num_pixel_issues:
+      Size of mini-batches to use for estimating the label issues.
+      To maximize efficiency, try to use the largest `batch_size` your memory allows.
+      
+    n_jobs: int, optional
+      For num_pixel_issues:
+      Number of processes for multiprocessing (default value = 1). Only used on Linux.
+      If `n_jobs=None`, will use either the number of: physical cores if psutil is installed, or logical cores otherwise.
+    
+    verbose : bool, optional
+      For num_pixel_issues:
+      Whether to suppress print statements or not.
+      
+    **kwargs:
+      downsample : int, optional
+      Factor to shrink labels and pred_probs by for 'num_pixel_issues' only . Default ``16``
+      Must be a factor divisible by both the labels and the pred_probs. Note that larger factors result in a linear 
+      decrease in performance
+
+
+
+    Returns
+    -------
+
+    image_scores: np.ndarray
+        Array of shape ``(N, )`` of scores between 0 and 1, one per image in the dataset.
+        Lower scores indicate image more likely to contain a label issue.
+    pixel_scores:
+        Array of shape ``(N,H,W)`` of scores between 0 and 1, one per pixel in the dataset.
+
+    """
+        
+    def check_input(labels: np.ndarray, pred_probs: np.ndarray) -> None:
+        if len(labels.shape) != 3:
+            raise ValueError("labels must have a shape of (N, H, W)")
+
+        if len(pred_probs.shape) != 4:
+            raise ValueError("pred_probs must have a shape of (N, K, H, W)")
+
+        num_images, height, width = labels.shape
+        num_images_pred, num_classes, height_pred, width_pred = pred_probs.shape
+
+        if num_images != num_images_pred or height != height_pred or width != width_pred:
+            raise ValueError("labels and pred_probs must have matching dimensions for N, H, and W")
+            
+    check_input(labels,pred_probs)
+    
+    softmin_temperature = kwargs.get("temperature", 0.1) 
+    downsample_num_pixel_issues = kwargs.get("downsample", 16) 
+
+    if method == "num_pixel_issues":
+        _,K,_,_ = pred_probs.shape
+        labels_expanded = labels[:, np.newaxis, :, :]
+        mask = (np.arange(K)[np.newaxis, :, np.newaxis, np.newaxis] == labels_expanded)
+        # Calculate pixel_scores
+        masked_pred_probs = np.where(mask, pred_probs, 0)
+        pixel_scores = masked_pred_probs.sum(axis=1)
+        
+        return find_label_issues(labels,pred_probs, downsample=downsample_num_pixel_issues, n_jobs=n_jobs, scores_only=True, verbose=verbose,), pixel_scores
+        
+    if downsample_num_pixel_issues != 16:
```

Comment:
Wouldn't this be triggered everytime even if `method=='num_pixel_issues'`? (did you mean to make this `else` instead of `if`?)

Reply:
No - I only want it triggered for 'num_pixel_issues' - it works as intended right now since it would skip over this if the setting is softmin